python/dask_cudf/dask_cudf/groupby.py

```diff
@@ -0,0 +1,345 @@
+        for j, split in enumerate(
+            gb.partition_by_hash(gb_cols, split_out, keep_index=False)
+        ):
+            output[j] = split
+    else:
+        # Dask-Dataframe (Pandas) support
+        output = hash_shard(
+            gb,
+            split_out,
+            split_out_setup=split_out_on_cols,
+            split_out_setup_kwargs={"cols": gb_cols},
+        )
+    del gb
+    return output
+
+
+def _tree_node_agg(dfs, gb_cols, split_out, dropna, sort, sep):
+    df = _concat(dfs, ignore_index=True)
+    agg_dict = {}
+    for col in df.columns:
+        if col in gb_cols:
+            continue
+        agg = col.split(sep)[-1]
+        if agg in ("count", "sum"):
+            agg_dict[col] = ["sum"]
+        elif agg in ("min", "max"):
+            agg_dict[col] = [agg]
+        else:
+            raise ValueError(f"Unexpected aggregation: {agg}")
+
+    gb = df.groupby(gb_cols, dropna=dropna, as_index=False, sort=sort).agg(
+        agg_dict
+    )
+
+    # Don't include the last aggregation in the column names
+    gb.columns = [_make_name(*name[:-1], sep=sep) for name in gb.columns]
+    return gb
+
+
+def _finalize_gb_agg(
+    gb,
+    gb_cols,
+    aggs,
+    columns,
+    final_columns,
+    as_index,
+    sort,
+    sep,
+    str_cols_out,
+):
+
+    # Deal with higher-order aggregations
+    for col in columns:
+        agg_list = aggs.get(col, [])
+        agg_set = set(agg_list)
+        if agg_set.intersection({"mean", "std", "var"}):
+            count_name = _make_name(col, "count", sep=sep)
+            sum_name = _make_name(col, "sum", sep=sep)
+            if agg_set.intersection({"std", "var"}):
+                n = gb[count_name]
+                x = gb[sum_name]
+                pow2_sum_name = _make_name(col, "pow2", "sum", sep=sep)
+                x2 = gb[pow2_sum_name]
+                result = x2 - x ** 2 / n
+                ddof = 1
+                div = n - ddof
+                div[div < 1] = 1
+                result /= div
+                result[(n - ddof) == 0] = np.nan
+                if "var" in agg_list:
+                    name_var = _make_name(col, "var", sep=sep)
+                    gb[name_var] = result
+                if "std" in agg_list:
+                    name_std = _make_name(col, "std", sep=sep)
+                    gb[name_std] = np.sqrt(result)
+                gb.drop(columns=[pow2_sum_name], inplace=True)
+            if "mean" in agg_list:
+                mean_name = _make_name(col, "mean", sep=sep)
+                gb[mean_name] = gb[sum_name] / gb[count_name]
+            if "sum" not in agg_list:
+                gb.drop(columns=[sum_name], inplace=True)
+            if "count" not in agg_list:
+                gb.drop(columns=[count_name], inplace=True)
+
+    # Ensure sorted keys if `sort=True`
+    if sort:
+        gb = gb.sort_values(gb_cols)
+
+    # Set index (use `inplace` when supported)
+    if as_index:
+        gb = gb.set_index(gb_cols)
```

Comment:
I know Pandas deep copies on `set_index` and I believe cuDF follows the semantics as well. Could we use `gb.set_index(gb_cols, inplace=True)` here instead?

Reply:
Yeah - Just need #6231 for this.